src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -61,6 +59,7 @@ public Spool(DeviceConfiguration deviceConfiguration, Kernel kernel) {
         topics.subscribe((what, node) -> {
             if (WhatHappened.childChanged.equals(what) && node != null) {
                 setSpoolerConfigFromDeviceConfig(topics);
+                spooler = setupSpooler();
```

Comment:
Thread safety? What's going on with the old spooler you're replacing, won't we lose messages?